minion/endtoend_consumer.go

```diff
@@ -0,0 +1,104 @@
+package minion
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/twmb/franz-go/pkg/kgo"
+	"github.com/twmb/franz-go/pkg/kmsg"
+	"go.uber.org/zap"
+)
+
+func (s *Service) ConsumeFromManagementTopic(ctx context.Context) error {
+	client := s.kafkaSvc.Client
+	topicMessage := s.Cfg.EndToEnd.TopicManagement.Name
+	topic := kgo.ConsumeTopics(kgo.NewOffset().AtEnd(), topicMessage)
+	balancer := kgo.Balancers(kgo.CooperativeStickyBalancer()) // Default GroupBalancer
+	switch s.Cfg.EndToEnd.Consumer.RebalancingProtocol {
+	case RoundRobin:
+		balancer = kgo.Balancers(kgo.RoundRobinBalancer())
+	case Range:
+		balancer = kgo.Balancers(kgo.RangeBalancer())
+	case Sticky:
+		balancer = kgo.Balancers(kgo.StickyBalancer())
+	}
+	client.AssignPartitions(topic)
+	client.AssignGroup(s.Cfg.EndToEnd.Consumer.GroupId, kgo.GroupTopics(topicMessage), balancer)
+	s.logger.Info("starting to consume topicManagement")
+
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		default:
+			startConsumeTimestamp := timeNowMs()
+			fetches := client.PollFetches(ctx)
+			errors := fetches.Errors()
+			for _, err := range errors {
+				// Log all errors and continue afterwards as we might get errors and still have some fetch results
+				s.logger.Error("failed to fetch records from kafka",
+					zap.String("topic", err.Topic),
+					zap.Int32("partition", err.Partition),
+					zap.Error(err.Err))
+			}
+
+			iter := fetches.RecordIter()
+			var record *kgo.Record
+			for !iter.Done() {
+				record = iter.Next()
+
+				if record == nil {
+					continue
+				}
+
+				res := TopicManagementRecord{}
+				json.Unmarshal(record.Value, &res)
+
+				// Push the latency to endtoendLatencies that will be consumed by prometheus later
+				latencySec := float64(timeNowMs()-res.Timestamp) / float64(1000)
+				s.observeLatencyHistogram(latencySec, int(record.Partition))
+				s.storage.markRecordConsumed(record)
+			}
+			uncommittedOffset := client.UncommittedOffsets()
+			// Only commit if uncommittedOffset returns value
+			if uncommittedOffset != nil {
+				startCommitTimestamp := timeNowMs()
+				client.CommitOffsets(ctx, uncommittedOffset, func(_ *kmsg.OffsetCommitRequest, _ *kmsg.OffsetCommitResponse, err error) {
+					if err != nil {
+						// "unable to dial the broker" is temporary error, won't be printed for cleanliness of the log
+						// https://pkg.go.dev/github.com/twmb/kafka-go@v0.4.9/pkg/kgo#pkg-variables
```

Comment:
This version of the package is pretty wildly out of date?

Reply:
Good point. Did focus on the latest tag on the docu.